addons/mail/static/src/model/store.js

```diff
@@ -132,16 +135,17 @@ export class Store extends Record {
                             }
                         }
                     }
+                    deletingRecordsByLocalId.set(record.localId, record);
+                    this.recordByLocalId.delete(record.localId);
                     this._.ADD_QUEUE("hard_delete", toRaw(record));
                 }
                 while (RHD_QUEUE.size > 0) {
                     // effectively delete the record
                     /** @type {Record} */
                     const record = RHD_QUEUE.keys().next().value;
                     RHD_QUEUE.delete(record);
+                    deletingRecordsByLocalId.delete(record.localId);
                     record._[IS_DELETED_SYM] = true;
```

Comment:
This should be moved too to fix exists (+ add test). And conceptually this flag should always be set together when removing record from recordByLocalId.

Reply:
Moved `IS_DELETED_SYM` so that this is set for soft-deleted records. This makes `.exists()` consistent with `.records` and `.get()` that don't return (soft) deleted records. I enriched 1st test to cover `.exists()` on soft-deleted record.

`IS_DELETING_SYM` is kept but flagged deprecated since this is 100% the same as `IS_DELETED_SYM`.